include/nuttx/sched_note.h

```diff
@@ -256,15 +267,31 @@ struct note_syscall_leave_s
 struct note_irqhandler_s
 {
   struct note_common_s nih_cmn; /* Common note parameters */
-  int nih_irq;                  /* IRQ number */
+  uint8_t nih_irq;              /* IRQ number */
```

Comment:
The change for note_irqhandler_s, note_syscall_leave_s and note_syscall_enter_s could move to another patch and apply to the master directly.

Reply:
I got it. I'll create another PR for it to the master.
And create the PR to cherry-pick 02c38e6 5b96c1e 09296d5.